File: codegen/results/v002/haiku/20260411_153257/typescript/001_csv_to_json/solution.ts
import fs from 'fs';
import path from 'path';

const filePath = path.join(process.cwd(), 'input', 'data.csv');
const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.trim().split('\n');
const rows = lines.slice(1).map(line => {
  const [Name, Age, Email, Score] = line.split(',');
  return {
    Name: Name.trim(),
    Age: parseInt(Age.trim(), 10),
    Email: Email.trim(),
    Score: parseFloat(Score.trim()),
  };
});
console.log(JSON.stringify(rows));

File: codegen/results/v002/haiku/20260411_153257/python/001_csv_to_json/solution.py
import csv
import json
import sys

rows = []
with open('input/data.csv', newline='') as f:
    reader = csv.DictReader(f)
    for row in reader:
        rows.append({
            'Name': row['Name'],
            'Age': int(row['Age']),
            'Email': row['Email'],
            'Score': float(row['Score'])
        })

print(json.dumps(rows))
